src/hog.jl

```diff
@@ -0,0 +1,99 @@
+"""
+```
+hog_params = HOG([orientations = 9], [cell_size = 8], [block_size = 2], [block_stride = 1], [norm_method = "L2-norm"])
+```
+
+Parameters:  
+-    orientations   = number of orientation bins
+-    cell_size      = size of a cell is cell_size x cell_size (in pixels)
+-    block_size     = size of a block is block_size x block_size (in terms of cells)
+-    block_stride   = stride of blocks. Controls how much adjacent blocks overlap.
+-    norm_method    = block normalization method. Options: L2-norm, L2-hys, L1-norm, L2-sqrt. 
+"""
+type HOG <: Params
+    orientations::Int 
+    cell_size::Int
+    block_size::Int
+    block_stride::Int 
+    norm_method::String
+end
+
+function HOG(; orientations::Int = 9, cell_size::Int = 8, block_size::Int = 2, block_stride::Int = 1, norm_method::String = "L2-norm")
+    HOG(orientations, cell_size, block_size, block_stride, norm_method)
+end
+
+function create_descriptor{T<:Images.NumberLike}(img::AbstractArray{T, 2}, params::HOG)
```

Comment:
There are various extensions of HOG to color.  The original Dalal and Triggs paper suggests computing the HOG in each channel separately, and choosing the value from the channel with the highest gradient magnitude.